libphal/phal_sbe.C

```diff
@@ -370,5 +398,541 @@ void threadStopProc(struct pdbg_target *proc)
+
+		if (sbeState == SBE_STATE_DEBUG_MODE ||
+		    sbeState == SBE_STATE_FAILED) {
+			// The SBE on this processor is not in a state
+			// where chipOps can be issued. SBE Error handling
+			// will be skipped.
+			log(level::ERROR,
+			    "SBERC: sbe for proc %s is not in DEBUG/FAILED "
+			    "state %d",
+			    pdbg_target_path(proc), sbeState);
+			throw sbeError_t(exception::SBE_INVALID_STATE);
+		}
+
+		// set state to debug mode to extract sbe rc
+		setState(proc, SBE_STATE_DEBUG_MODE);
+
+		// Execute SBE extract rc to set up sdb bit for pibmem dump to
+		// work
+		fapi2::ReturnCode fapiRc;
+		P10_EXTRACT_SBE_RC::RETURN_ACTION recovAction;
+
+		// p10_extract_sbe_rc is returning the error along with
+		// recovery action, so not checking the fapirc.
+		fapiRc = p10_extract_sbe_rc(proc, recovAction, true);
+		if (fapiRc != fapi2::FAPI2_RC_SUCCESS) {
+			log(level::INFO,
+			    "SBERC: p10_extract_sbe_rc returned error %d, skip "
+			    "recovery",
+			    fapiRc);
+			return;
+		}
+		log(level::INFO,
+		    "SBERC: p10_extract_sbe_rc for proc=%s returned rc=0x%08X "
+		    "and "
+		    "SBE Recovery Action=0x%08X",
+		    pdbg_target_path(proc), fapiRc, recovAction);
+
+		// marking sbe state failed after p10_extract_sbe_rc so
+		// that no further recovery actions are executed in parallel
+		setState(proc, SBE_STATE_FAILED);
+
+		switch (recovAction) {
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::RESTART_SBE:
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::RESTART_CBS:
+			log(level::INFO,
+			    "SBERC: recovery action RESTART_SBE/RESTART_CBS "
+			    "received "
+			    "on Proc %s",
+			    pdbg_target_path(proc));
+			(void)restartSBERCHandler(proc);
+			break;
+
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::REIPL_BKP_SEEPROM:
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::REIPL_UPD_SEEPROM:
+			log(level::INFO,
+			    "SBERC: recovery action "
+			    "REIPL_BKP_SEEPROM/REIPL_UPD_SEEPROM "
+			    "received on Proc %s",
+			    recovAction, pdbg_target_path(proc));
+			(void)reiplUsingBKPBootSeeprom(proc);
+			break;
+
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::REIPL_BKP_MSEEPROM:
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::REIPL_UPD_MSEEPROM:
+			log(level::INFO,
+			    "SBERC: recovery action "
+			    "REIPL_BKP_MSEEPROM/REIPL_UPD_MSEEPROM "
+			    "received on Proc %s",
+			    pdbg_target_path(proc));
+			(void)reiplUsingBKPMeasurementSeeprom(proc);
+			break;
+
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::NO_RECOVERY_ACTION:
+			log(level::INFO,
+			    "SBERC: recovery action NO_RECOVERY_ACTION "
+			    "received on "
+			    "Proc %s",
+			    pdbg_target_path(proc));
+			(void)reiplUsingBKPBootSeeprom(proc);
+			break;
+
+		case P10_EXTRACT_SBE_RC::RETURN_ACTION::REIPL_BKP_BMSEEPROM:
+			log(level::INFO,
+			    "SBERC: recovery action REIPL_BKP_BMSEEPROM "
+			    "received on "
+			    "Proc %s",
+			    pdbg_target_path(proc));
+			(void)reiplUsingBKPBootSeeprom(proc);
+			(void)reiplUsingBKPMeasurementSeeprom(proc);
```

Comment:
c++ casting style instead of C style casting like static_cast<void>(reiplUsingBKPBootSeeprom(proc)); in places of (void)reiplUsingBKPBootSeeprom(proc);

Reply:
here I am indicating that the method returns void, I am not doing any casting , I am just calling the method.